projects/js-packages/components/components/pricing-slider/index.tsx

```diff
@@ -22,7 +22,27 @@ const PricingSlider: React.FC< PricingSliderProps > = ( {
 	onAfterChange,
 	renderThumb,
 } ) => {
-	const componentClassName = classNames( 'jp-components-pricing-slider', className );
+	const [ isThumbHolding, setIsThumbHolding ] = React.useState( false );
+
+	const componentClassName = classNames( 'jp-components-pricing-slider', className, {
+		'jp-components-pricing-slider--is-holding': isThumbHolding,
+	} );
+
+	const onBeforeChangeCallback = beforeValue => {
+		setIsThumbHolding( true );
+
+		if ( onBeforeChange && typeof onBeforeChange === 'function' ) {
+			onBeforeChange( beforeValue );
+		}
+	};
+
+	const onAfterChangeCallback = afterValue => {
+		setIsThumbHolding( false );
+
+		if ( onAfterChange && typeof onAfterChange === 'function' ) {
+			onAfterChange( afterValue );
+		}
```

Comment:
```suggestion
		if ( typeof onBeforeChange === 'function' ) {
			onBeforeChange( beforeValue );
		}
	};

	const onAfterChangeCallback = afterValue => {
		setIsThumbHolding( false );

		if ( typeof onAfterChange === 'function' ) {
			onAfterChange( afterValue );
		}
```

`typeof` accepts undefined operands without throwing an exception.